src/averageAges.js

```diff
@@ -15,11 +15,16 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  let menFiltered = people.filter(i => i.sex === 'm');
+
+  menFiltered = century
+    ? menFiltered.filter(person => Math.ceil(person.died / 100) === century)
+    : menFiltered;
+
+  const menAges = menFiltered.map((person) => person.died - person.born);
+  const averageAgeMen = menAges.reduce((sum, x) => sum + x, 0) / menAges.length;
```

Comment:
no need to create a new array with map, find `person.died - person.born` inside reduce method